datacommons_client/endpoints/node.py

```diff
@@ -252,99 +253,233 @@ def fetch_entity_names(
 
     return names
 
+  def _fetch_contained_in_place(
+      self,
+      node_dcids: str | list[str],
+      out: bool = True,
+      contained_type: Optional[str] = None,
+      as_dict: bool = False,
+  ) -> dict[str, list[Node | dict]]:
+    """Fetches places that contain or are contained in the given nodes. Uses the
+          `containedInPlace` property to fetch parent or child place relationships.
+
+        Args:
+            node_dcids (str | list[str]): One or more DCIDs representing geographic places.
+            out (bool, optional): If True, fetch places contained in the given node(s).
+                If False, fetch places that contain the given node(s). Defaults to True.
+            contained_type (str, optional): Optional type constraint (e.g., 'Country',
+                'Country'). If provided, only fetches places of that type.
+            as_dict (bool, optional): If True, returns the result as a dictionary of
+                lists of dictionaries. If False, returns Node objects. Defaults to False.
+
+        Returns:
+            dict[str, list[dict]] | dict[str, list[Any]]: A dictionary where keys are DCIDs
+            and values are lists of place relationships, either as raw objects or
+            dictionaries (if `as_dict` is True).
+        """
+    data = self.fetch_property_values(
+        node_dcids=node_dcids,
+        properties=("containedInPlace+"
+                    if contained_type else "containedInPlace"),
+        out=out,
+        constraints=f"typeOf:{contained_type}" if contained_type else None,
+    ).get_properties()
+
+    if as_dict:
+      for entity, nodes in data.items():
+        if not nodes:
+          continue
+        if isinstance(nodes, Node):
+          return {entity: [nodes.to_dict()]}
+        else:
+          return {entity: [node.to_dict() for node in nodes]}
```

Comment:
Since I merged https://github.com/datacommonsorg/api-python/pull/244, using as_dict=True is broken now. For example:

```
_fetch_contained_in_place(dc_client.node, "geoId/06", contained_type="County", out=False, as_dict=True)
```

raises a AttributeError: 'str' object has no attribute 'to_dict' on this line.

This is because `get_properties` returns Dict[NodeDCID, Dict[Property, List[Node]]] so you'd need to provide the property. See suggestion below

Reply:
Thank you for the fix!